dialer.go

```diff
@@ -153,16 +165,81 @@ func WithConnectionProperties(props amqp.Table) Option {
 	}
 }
 
+func WithReadyChan(receiver chan struct{}) Option {
+	return func(c *Dialer) {
+
+		if receiver == nil {
+			receiver = make(chan struct{}, 1)
+		} else {
+			if cap(receiver) == 0 {
+				panic(errChanIsUnbuffered)
+			}
+		}
+
+		c.notifyReady = append(c.notifyReady, receiver)
+	}
+}
+
+func WithUnreadyChan(receiver chan error) Option {
+	f := func(c *Dialer) {
+
```

Comment:
in `WithUnreadyChan` you should append channel as is, without any additional checks. Later, [after you've applied options](https://github.com/makasim/amqpextra/blob/50be62389638a2a4dbc41c5f9ecf907890ea1c04/dialer.go#L98) you can check for nil and cap and return an error.

Reply:
same applies to readyCh

Reply:
+

Reply:
checks removed